CONTRIBUTING.md

```diff
@@ -0,0 +1,141 @@
+- Naming scheme
+- `ActorFunc` naming specifics
+- Actor `struct` & `enum` defintions go in the header file
+- Renaming symbols referenced by asm (?)
+- Documentation step - what does this entail? (Not required for actors, but if started should be complete. Should be required for code/boot)
+
+### `NON_MATCHING`
+
+If you make significant progress on decompiling a function, but can't get it to match perfectly, you can use a `NON_MATCHING` block to commit your code but have it excluded from the build, like so:
+
+```c
+#ifdef NON_MATCHING
+void CollisionCheck_SpawnWaterDroplets(GlobalContext* ctxt, Vec3f* v) {
+    // ... 
+}
+#else
+#pragma GLOBAL_ASM("./asm/non_matchings/code/z_collision_check/CollisionCheck_SpawnWaterDroplets.asm")
+#endif
+```
+
+### Matching vs. Documenting
+
+Usually, the first step of decompiling a section of code is to get it *matching*: to produce a C version of the code that can be compiled into an identical ROM.
+
+However, the goal of this project is to produce a codebase that can be understood and modified.
+So, beyond producing matching C code, the next steps are *documenting* the code.
+
+Documenting is more than just adding comments, and entails:
+
+- Renaming functions, variables, and struct members
+- Using (or adding) constants and macros when possible
+- Commenting sections of code that are not straightforward
+- (TODO)
+
+It is expected that additions to `boot/` and `code/` are fully documented.
+Overlays do not need to be fully documented, but it is encouraged.
+
+Pull Requests (PRs)
+-------------------
+
+### Checklist
+
+Before opening a PR, walk through the following steps to ensure that your code confroms to the style guide & conventions.
+
+- Run `./format.sh` to reformat your updated files.
+- Ensure `make` successfully builds a matching ROM.
+- Your changes should not introduce new warnings (TODO: See PR #150 for check)
+- New variables & functions should follow the naming conventions above.
+- Check spelling in comments & renamed symbols
+
+### Pull Request Process
+
+TODO: What should go in the pull request body?
+
+After opening a PR, the Jenkins server will verify your changes.
+If there is an error, double-check that you can successfully `make clean && make` locally and that all added/modified files were `git add`-ed to your commit.
+
+Each PR needs a review from one reviewer, plus the project lead.
+
+The PR author marks the comments as resolved, since they are the ones with permissions.
+Once all comments are addressed, it is courteous to ping the reviewer on either Discord or GitHub via re-requesting a review.
+
+After all the comments are addressed and at least one contributor has approved the review, the project lead can then merge the code.
+The project lead is also responsible for ensuring that all of these procedures are followed.
+
+Throughout the process, you should update the [Trello board](https://trello.com/b/ruxw9n6m/majoras-mask-decompilation) with the status of your PR.
+
+Tools
```

Comment:
Remove this and add a link to tools.md

Reply:
I removed the tools section; but I think we decided not to write a tools.md in this PR. (So I'm not going to mention anything here right now)